src/pytest_plugins/test_filler/test_filler.py

```diff
@@ -263,8 +263,6 @@ def get_module_dir(item) -> str:
         if not m:
             raise Exception("Could not parse test name: " + item.name)
         name = m.group(1)
-        if fixture.name:
-            name += "-" + fixture.name
         jsonFixture = fixture.to_json()
         self.all_fixtures[module_dir].append((name, jsonFixture))
```

Comment:
I think removing L266-267 has removed the test function name from the filler name. But, we did briefly discuss in our weekly that we could change the `name` to be the test's pytest Item `nodeid`. 

This would mean deleting L265-L268 and changing (tested locally):

```suggestion
        self.all_fixtures[module_dir].append((item.nodeid, jsonFixture))
```

The motivation for this change is as following: The `nodeid` can be directly provided to fill to execute this test case in isoloation, e.g.

```
# get the nodeid's using --collect-only -q
✦1 ➜ fill --collect-only -q -k test_warm_coinbase_call_out_of_gas
tests/shanghai/eip3651_warm_coinbase/test_warm_coinbase.py::test_warm_coinbase_call_out_of_gas[fork=Shanghai-CALL-sufficient_gas]
tests/shanghai/eip3651_warm_coinbase/test_warm_coinbase.py::test_warm_coinbase_call_out_of_gas[fork=Shanghai-CALL-insufficient_gas]
tests/shanghai/eip3651_warm_coinbase/test_warm_coinbase.py::test_warm_coinbase_call_out_of_gas[fork=Shanghai-CALLCODE-sufficient_gas]
tests/shanghai/eip3651_warm_coinbase/test_warm_coinbase.py::test_warm_coinbase_call_out_of_gas[fork=Shanghai-CALLCODE-insufficient_gas]
tests/shanghai/eip3651_warm_coinbase/test_warm_coinbase.py::test_warm_coinbase_call_out_of_gas[fork=Shanghai-DELEGATECALL-sufficient_gas]
tests/shanghai/eip3651_warm_coinbase/test_warm_coinbase.py::test_warm_coinbase_call_out_of_gas[fork=Shanghai-DELEGATECALL-insufficient_gas]
tests/shanghai/eip3651_warm_coinbase/test_warm_coinbase.py::test_warm_coinbase_call_out_of_gas[fork=Shanghai-STATICCALL-sufficient_gas]
tests/shanghai/eip3651_warm_coinbase/test_warm_coinbase.py::test_warm_coinbase_call_out_of_gas[fork=Shanghai-STATICCALL-insufficient_gas]

1/228 tests collected (227 deselected) in 0.19s
```
and then execute a single test via its nodeid:
```
fill tests/shanghai/eip3651_warm_coinbase/test_warm_coinbase.py::test_warm_coinbase_call_out_of_gas[fork=Shanghai-CALL-sufficient_gas]
```

This will result in fixture names:
```
{
    "000-tests/shanghai/eip3651_warm_coinbase/test_warm_coinbase.py::test_warm_coinbase_call_out_of_gas[fork=Shanghai-CALL-sufficient_gas]": {
        "_info": {
            "filling-transition-tool": "evm version 1.13.0-unstable-fd389791-20230926",
            "reference-spec": "https://github.com/ethereum/EIPs/blob/master/EIPS/eip-3651.md",
            "reference-spec-version": "d94c694c6f12291bb6626669c3e8587eef3adff1"
        },
        "network": "Shanghai",
        ....
```

This is strictly not a refactor, but perhaps we can sneak it in here? :slightly_smiling_face: cc @marioevz 

Do we need counter prefix here? `000-`, `001-`,... in the fixture name?

Reply:
I like this a lot Dan 💯 

My vote is to remove the `0XX-` prefix :D

Reply:
Correct me if I'm wrong, but I think we could even leave it as "test_warm_coinbase_call_out_of_gas[fork=Shanghai-CALL-sufficient_gas]" because the "tests/shanghai/eip3651_warm_coinbase/test_warm_coinbase.py" part is redundant since it can be deduced from the file location.

We could make a dry run and test that there are no files with two different tests using the same name just to be sure that this enough. What do you think?

Also if we do this, on pyspecs sim in hive, the name of the test must now include the file path to make sure two tests do not have the same name.

Reply:
I understand the desire to keep this minimal, but the motivation to use nodeid is because, the following doesn't work:
```
fill test_warm_coinbase_call_out_of_gas[fork=Shanghai-CALL-sufficient_gas]
```
-> 
```
ERROR: file or directory not found: test_warm_coinbase_call_out_of_gas[fork=Shanghai-CALL-sufficient_gas]
```
It only seems to work with the full nodeids. Note, this is not using `-k`.